core/controllers/admin_test.py

```diff
@@ -1785,15 +1778,12 @@ def test_handler_with_invalid_username(self) -> None:
 
         self.login(self.SUPER_ADMIN_EMAIL, is_super_admin=True)
         csrf_token = self.get_new_csrf_token()
-        response = self.put_json(
+        self.put_json(
             '/topicmanagerrolehandler', {
                 'action': 'assign',
                 'username': username,
                 'topic_id': topic_id
-            }, csrf_token=csrf_token, expected_status_int=400)
-
-        self.assertEqual(
-            response['error'], 'User with given username does not exist.')
+            }, csrf_token=csrf_token, expected_status_int=404)
```

Comment:
I think we should test the error message coming from the response to validate that the message that is coming is expected, right? This will help us in making sure that we send the correct response to frontend.
Same goes for below mentioned tests.

Reply:
Thanks for the quick review. I deleted the error message since the error message will be kind of fixed to `Could not find the page xxhandler`and not related to the exception message hard coded in admin.py as handled in https://github.com/oppia/oppia/blob/bae50329930f60f8a5a9191fd1cc89165f8f1006/core/controllers/base.py#L805-L813. Also the error class `PageNotFoundException` that we use is the only error class for handling status code 404, so I think the status verification should be sufficient. 

I also found similar unit test of only checking status code for `PageNotFoundException` in https://github.com/oppia/oppia/blob/develop/core/controllers/skill_editor_test.py#L116-L122. 

Let me know if this sounds good to you.

Reply:
@lkbhitesh07 Sorry that I didn't figure out how to assign this to you. I could unassign myself but wasn't able to assign to anyone else.

Reply:
I see, thanks for the explanation. I am still more tilted toward displaying the cause of the error as the current error response is a bit unclear and might confuse the user. I am wondering if there is a way we can opt where we can send the correct error response, I think we should ask @seanlip's suggestion as mainly Sean will be using the admin page.
Thanks

Reply:
If it's easy to test the error message then I think we might as well, yeah.